src/vos/vos_ts.h

```diff
@@ -497,8 +497,19 @@ vos_ts_set_add(struct vos_ts_set *ts_set, uint32_t *idx, const void *rec,
 	}
 
 calc_hash:
-	if (ts_set->ts_etype > VOS_TS_TYPE_CONT)
-		hash = vos_hash_get(rec, rec_size);
+	if (ts_set->ts_etype > VOS_TS_TYPE_CONT) {
+		if (ts_set->ts_etype != VOS_TS_TYPE_OBJ) {
+			hash = vos_hash_get(rec, rec_size);
+		} else {
+			daos_unit_oid_t *oid = (daos_unit_oid_t *)rec;
+
+			/*
+			 * Test shows using d_hash_murmur64() for oid
+			 * is easy to conflict.
+			 */
+			hash = oid->id_pub.lo ^ oid->id_pub.hi;
```

Comment:
So the key point here is to ignore the 'id_shard' or changing murmur to ^ operation? did you try murmur for daos_obj_id_t part only?

Reply:
I think it is changing murmur to ^ operation, i can try murmur for daos_obj_id_t part only.

Reply:
Just tested trying murmur for daos_obj_id_t part will result loops to 316 but still not better than ^ operation directly.

Reply:
btw, do you think i should xor ^ id_shard as well? i think shard could be extracted from daos_obj_id_t, not sure why an extra id_shard is needed here.

Reply:
I'm not quite sure 'id_shard' should be included in hash or not, but seems it's already improved a lot by XOR daos_obj_id_t only, so I'm find with landing it as-is.

Reply:
I'm actually trying to reason how you can get more than 18K entries without a collision when the negative cache for objects is only 16K entries

Reply:
false alarm, there are 64K entries